### PR TITLE
feat: add 8-bit DEC n instruction with unit tests

### PR DESCRIPTION
**Pull request by marcluque**
Add DEC n instruction and unit tests. Advances progress on issue https://github.com/Benzammour/yobemag/issues/12.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -86,7 +86,7 @@ PenaltyIndentedWhitespace: 0
 PointerAlignment: Right
 
 QualifierAlignment: Custom
-QualifierOrder: ['inline', 'static', 'restrict', 'type', 'const', 'volatile' ]
+QualifierOrder: [ 'inline', 'static', 'const', 'volatile', 'restrict', 'type' ]
```
Comment (Benzammour) [inline .clang-format:89]: Why the change in order?

Comment (marcluque) [reply .clang-format:89]: Mainly because of type being the right-most. Such that we do not have something like `int const *`, but instead `const int *`. The rest is somewhat arbitrary/up for debate.

**Review by Benzammour — APPROVED** [review]
